src/components/stimuli/Stimuli.tsx

```diff
@@ -29,15 +41,104 @@ const Stimuli: React.FC = () => {
       }
     }
 
-    fetchStimuliUrls();
-  }, []);
+    if (!isUploading) fetchStimuliUrls();
+  }, [isUploading, file]);
+
+  const fileChangedHandler = (event: any) => {
+    setFileInputText(event.target.files[0].name);
+    setFile(event.target.files[0]);
+  };
+
+  const handleUploadClick = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (file) {
+      const formData = new FormData();
+      formData.append('file', file);
+      setUploading(true);
+      const response = await fetch(
+        `${process.env.REACT_APP_BACKEND_ADDRESS}/image`,
+        {
+          method: 'POST',
+          body: formData,
+        }
+      );
+      const body = await response.json();
+      setResponseMessage(body.message);
+      setUploading(false);
+      setFile('');
+      setFileInputText(DEFAULT_FILE_INPUT_TEXT);
+      setShowToast(true);
+    }
+  };
 
   return (
     <Container>
       <Row>
+        <Col>
+          <Form onSubmit={handleUploadClick}>
+            <div className="input-group">
+              <div className="input-group-prepend">
+                <Button
+                  variant="secondary"
+                  type="submit"
+                  disabled={file === '' || isUploading}
+                >
+                  {isUploading ? (
+                    <Spinner
+                      as="span"
+                      animation="border"
+                      size="sm"
+                      role="status"
+                      aria-hidden="true"
+                    />
+                  ) : (
+                    'Upload Image'
```

Comment:
```suggestion
                    'Upload'
```

Recommended for consistency with Experiment upload page.